pkg/gcp/ocpgwdeployer.go

```diff
@@ -249,17 +331,41 @@ func (d *ocpGatewayDeployer) Cleanup(reporter api.Reporter) error {
 		}
 
 		for _, instance := range instanceList.Items {
+			if !d.isInstanceGatewayNode(instance) {
+				continue
+			}
+
+			// If the instance name matches with d.gcp.infraID + "-submariner-gw-" + zone.Name, it implies that
+			// the gateway node was deployed using the OCPMachineSet API otherwise it's an existing worker node.
 			prefix := d.gcp.infraID + "-submariner-gw-" + zone.Name
 			if strings.HasPrefix(instance.Name, prefix) {
+				reporter.Started(fmt.Sprintf("Deleting the Gateway instance %q", instance.Name))
 				err := d.deleteGateway(zone.Name)
+				if err != nil {
+					return reportFailure(reporter, err, "failed to delete dedicated gateway instance %q", instance.Name)
+				}
+
+				reporter.Succeeded("Successfully deleted the instance")
```

Comment:
I notice we're using a mix of message string constants and inlining the strings. I think we should be consistent. I prefer inlining as you've done here.

Reply:
I was trying to follow the existing pattern used in the existing AWS code - https://github.com/submariner-io/cloud-prepare/blob/579b81efb86d10ed7c89bc46d6bd9ae4b1a72d85/pkg/aws/aws.go#L25

But even I had the same feeling. After this PR is merged, I'll push a separate PR and use the inline string. I hope you are okay if I do it in a separate PR.